quickstart/sources.go

```diff
@@ -0,0 +1,172 @@
+package quickstart
+
+import (
+	"fmt"
+	"net/url"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/huh"
+	"github.com/pkg/errors"
+	"github.com/speakeasy-api/sdk-gen-config/workflow"
+	"github.com/speakeasy-api/speakeasy/charm"
+)
+
+func sourceBaseForm(inputWorkflow *workflow.Workflow) (*State, error) {
+	source := &workflow.Source{}
+	var sourceName string
+	if _, err := tea.NewProgram(charm.NewForm(huh.NewForm(
+		huh.NewGroup(
+			huh.NewInput().
+				Title("A name for this source:").
+				Placeholder("unique name across this workflow").
+				Prompt(" ").
+				Inline(true).
+				Validate(func(s string) error {
+					if _, ok := inputWorkflow.Sources[s]; ok {
+						return fmt.Errorf("a source with the name %s already exists", s)
+					}
+					return nil
+				}).
+				Value(&sourceName),
+		)),
+		"Let's setup a new source for your workflow.",
+		"A source is a compiled set of OpenAPI specs and overlays that are used as the input for a SDK generation.")).
+		Run(); err != nil {
+		return nil, err
+	}
+
+	promptForDocuments := true
+	for promptForDocuments {
+		var err error
+		sourceDocument, err := promptForDocument("input")
+		if err != nil {
+			return nil, err
+		}
+
+		source.Inputs = append(source.Inputs, *sourceDocument)
+
+		promptForDocuments, err = charm.NewBranchCondition("Would you like to add another input document?")
```

Comment:
Im not convinced still that we should support this in quickstart. It feels like needless extra steps to make users answer questions about mulitple docs and overlays when 99% of people will answer `no`. If we want to maintain this, I'd propose we have a flag like `speakeasy quickstart --advanced` or something. So most people wont see these questions, but if a customer does have a more complicated setup we can point them to that command

Reply:
The two below seems like more of a meta question on how much of speakeasy functionality we want quickstart to show. I'm not a huge fan of quickstart --advanced. I would rather push this into configure.


I'll list it at the top of PR. I want to here some of other feedback on whether people think we should be exposing things like multiple documents in quickstart or whether we should hide that in quickstart and wait for configure

Reply:
Discussed we will move this too conifgure